internal/decryption/header.go

```diff
@@ -41,17 +49,57 @@ func decodeHeader(logLine *AuditLogLine, opts KeyProviderOpts) (*HeaderRecord, e
 		return nil, err
 	}
 	return &HeaderRecord{
-		Timestamp:       logLine.TS,
-		Version:         logLine.Version,
-		CompressionMode: CompressionMode(logLine.CompressionMode),
+		Timestamp:       *logLine.TS,
+		Version:         *logLine.Version,
+		CompressionMode: CompressionMode(*logLine.CompressionMode),
 		KeyProvider:     keyProvider,
-		IV:              logLine.EncryptedKey[:16],
-		EncryptedLEK:    logLine.EncryptedKey[16:48],
-		AESBlock:        logLine.EncryptedKey[48:],
+		EncryptedBlock: HeaderEncryptedBlock{
+			IV:    logLine.EncryptedKey[:16],
+			LEK:   logLine.EncryptedKey[16:48],
+			Block: logLine.EncryptedKey[48:],
+		},
+		MAC: *logLine.MAC,
 	}, nil
 }
 
+func validateHeaderFields(logLine *AuditLogLine) error {
+	if logLine.TS == nil {
+		return fmt.Errorf("missing timestamp")
+	}
+
+	if logLine.Version == nil {
+		return fmt.Errorf("missing version")
+	}
+
+	if logLine.CompressionMode == nil {
```

Comment:
[q] is there a list of compression modes we support? should we validate the mode as well here?

Reply:
also, the spec says that it will be either "zstd" or not set, does that mean it's OK to be nil?

Reply:
The reason why it can be nil it is because we parse from a JSON and JSONs can miss keys. this check is to enforce presence of a value before using this field.

Reply:
Added validation